### ISSUE TITLE
Yarn global packages are installed unproperly

### ISSUE THREAD

**Issue by exarus** [ScoopInstaller/Scoop#1641]
Yarn that was installed via Scoop doesn't install global packages properly.

### TL;DR: Suggested hot fix workaround
If you are a Scoop user and want to fix the bug **ASAP** than you may add `%LOCALAPPDATA%\Yarn\config\global\node_modules\.bin` to your `PATH` environment variable.

### Test scenario:
I have Windows 10 desktop with fresh Scoop installed. After that:
```sh
scoop install nodejs yarn
yarn global add http-server
```
The last command generated file `$env:HOMEPATH\scoop\apps\yarn\current\bin\http-server.cmd`.
Scoop previously added this file's folder to `$env:PATH`.
This file contains only 1 non-empty line:
```bat
@"%~dp0\..\..\..\..\AppData\Local\Yarn\config\global\node_modules\.bin\http-server.cmd"   %*
```
The command `http-server` returns following output:
```
The system cannot find the path specified.
```

### Bug source
I've tried to find the bug reason. So I've edited the file and added 1 more `..\` right before AppData:
```bat
@"%~dp0\..\..\..\..\..\AppData\Local\Yarn\config\global\node_modules\.bin\http-server.cmd"   %*
```
The resulting file works.

The same issue is reproduced with any global Yarn package. We have to figure out if this a Scoop or Yarn issue.

**Comment by deevus** [comment]
@Daniel15 are there any known issues with `yarn global` on Windows?

**Comment by r15ch13** [comment]
I have installed scoop to `D:\scoop` which results in: 
```bat
@"%~dp0\C:\Users\r15ch\AppData\Local\Yarn\config\global\node_modules\.bin\http-server.cmd"   %*
```

**Comment by r15ch13** [comment]
It has something to do with `prefix` in the `%userprofile%\.yarnrc`.
Can also be reproduced by 
`yarn global add http-server --prefix "C:\\Users\\exarus\\scoop\\persist\\yarn\\bin"`
`yarn global add http-server --prefix "D:\\scoop\\persist\\yarn\\bin"`

I stumbled over this issue at the time I added the prefix to `.yarnrc`. See comments on https://github.com/lukesampson/scoop/commit/e4cc2c3c3115a51b6cba60a652c7ee66b85a9b1b

**Comment by Daniel15** [comment]
> I have installed scoop to D:\scoop which results in:

@r15ch13 - This is a known bug with `cmd-shim`: https://github.com/npm/cmd-shim/issues/21

**Comment by exarus** [comment]
@r15ch13, @Daniel15 The bug with different drives might be related to `cmd-shim`. But the bug with the invalid resolved path on the same drive is not a `cmd-shim` issue.

**Comment by Daniel15** [comment]
It might be due to how cmd-shim is being invoked by Yarn. Does Scoop use symlinks? Perhaps Yarn is doing strange things with symlinks on Windows.

**Comment by deevus** [comment]
@Daniel15 Yes the currently installed version is symlinked to `~/scoop/apps/yarn/current`

**Comment by dsbert** [comment]
I'm running into this issue as well. Typing `yarn global bin` outputs `C:\Users\usr\scoop\persist\yarn\bin\bin` for me. I do see the globally installed modules in that folder.

Checking my path variable, the following are both present.

```
C:\Users\usr\scoop\apps\yarn\current\Yarn\bin
C:\Users\usr\scoop\apps\yarn\current\bin
```

**Comment by mister-glass** [comment]
I have a similar issue but only while using git bash, in the cmd shell it seems to work.  (Windows 10)
I installed yarn 1.5.1 using scoop, then I installed eslint globally using yarn:

MINGW64:
```
$ eslint
/bin/sh: /c/Users/mrglass/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/eslint: No such file or directory
```

CMD:
```
C:\Users\mrglass\Desktop>eslint
eslint [options] file.js [file.js] [dir]
...
```

**Comment by akuntsch** [comment]
Same issue here with Git Bash. Works fine in cmd and PowerShell.

```
$ vue
/bin/sh: /c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/vue: No such file or directory
```

I think this might be an issue with Git Bash rather than scoop though, since `ls` also fails (on the file and the `.bin` directory):

```
$ ls /c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/vue
ls: cannot access '/c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/vue': No such file or directory
$ ls /c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/
ls: cannot access '/c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/': No such file or directory
```

However, `cd` does work on the `.bin` directory and the file obviously exists:
```
$ cd /c/Users/user/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/
$ ls
vue*  vue.cmd
```

**Comment by akuntsch** [comment]
After digging around a bit more I think this is an issue with symlinks from scoop (`bin` is persisted by scoop and the path depth of the persisted directory `scoop/persist/yarn/bin` is lower than the depth of the 'normal' path `scoop/apps/yarn/current/bin`) and yarn using relative directories in its shell scripts:
```
$ cat /c/Users/user/scoop/apps/yarn/current/bin/vue
#!/bin/sh
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")

case `uname` in
    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
esac

if [ -x "$basedir//bin/sh" ]; then
  "$basedir//bin/sh"  "$basedir/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/vue" "$@"
  ret=$?
else
  /bin/sh  "$basedir/../../../../../AppData/Local/Yarn/Data/global/node_modules/.bin/vue" "$@"
  ret=$?
fi
exit $ret
```

`$basedir` is probably resolved to the persisted `bin` directory, so `$basedir/../../../../..` will resolve to the Windows Users directory (e.g. `C:\Users`) instead of the user directory of the current user (e.g. `C:\Users\user`).

This could normally be fixed by setting the `global-folder` with `yarn config set` to something like `scoop/apps/yarn/current/global` but this does not work correctly (see https://github.com/yarnpkg/yarn/issues/5746 and https://github.com/yarnpkg/yarn/issues/5637). 

Not sure how to go about this. Editing the `.yarnrc` file in `post_install` like in https://github.com/yarnpkg/yarn/issues/5637 doesn't sound like the best way to do it (it does work though), especially since the file is managed by yarn and the changes might be overridden later on.

I guess the only 'good' workaround is to use `npm` for global installs until the `global-folder` config does actually work in yarn.

**Comment by h404bi** [comment]
I have a variant manifest of yarn: [dorado/yarn.json](https://github.com/h404bi/dorado/blob/master/yarn.json), which doesn't change the location of (or persist) yarn's global bin, I think you should try it. Previously I suggested to keep yarn's original location of global bin in https://github.com/lukesampson/scoop/pull/1862, but it failed, so I create the variant one.

**Comment by DanielRuf** [comment]
> @r15ch13 - This is a known bug with `cmd-shim`: [npm/cmd-shim#21](https://github.com/npm/cmd-shim/issues/21)

Well, the last commit was 3 years ago. Do we still use this outdated package for Yarn?

**Comment by DanielRuf** [comment]
> If you are a Scoop user and want to fix the bug **ASAP** than you may add `%LOCALAPPDATA%\Yarn\config\global\node_modules\.bin` to your `PATH` environment variable.

The correct path is now `%LOCALAPPDATA%\Yarn\Data\global\node_modules\.bin`.
config => Data

**Comment by masaeedu** [comment]
@DanielRuf Is this the same path that would be returned by `yarn global bin`? On the system I'm testing with (Win 10, yarn 1.12.3) I get `~\AppData\Local\Yarn\bin` instead.

**Comment by DanielRuf** [comment]
> ~\AppData\Local

This is the same, `%LOCALAPPDATA%` is an alias.

**Comment by masaeedu** [comment]
@DanielRuf Yes, I'm aware of that, but the last part of the part differs. The suffix I'm seeing to `%LOCALAPPDATA%\Local\Yarn` is `bin`, whereas in the path you provided it's `Data\global\node_modules\.bin`. I'm not sure which one is correct.

**Comment by DanielRuf** [comment]
Should be the same.

**Comment by niheaven** [comment]
It's fixed in https://github.com/lukesampson/scoop/pull/3206 (Yarn 1.15.2), please update and try again.

PS. the `yarn global bin` is left alone, that dir is not add to `$env:PATH` and is needless for scoop installed yarn.

**Comment by JustinGrote** [comment]
@niheaven this is still not fixed for me, it appears prefix still needs to be added or some yarn installs like azure-functions-core-tools still end up in %localappdata%.

As the bucket has moved to its own repository I willl reopen the issue there.

**Comment by doublethinkio** [comment]
Executing a command in PowerShell can quickly set environment variables to alleviate this problem
```powershell
$env:PATH="$env:PATH;$env:LOCALAPPDATA\Yarn\bin"
[Environment]::SetEnvironmentVariable("PATH", "$env:PATH", "User")
```
